plugin.rb

```diff
@@ -497,4 +497,40 @@ def self.enqueue_assign_hooks(event, payload)
     end
   end
 
+  register_search_advanced_filter(/in:assigned/) do |posts|
+    if @guardian.can_assign?
+      posts.where("topics.id IN (
+        SELECT tc.topic_id
+        FROM topic_custom_fields tc
+        WHERE tc.name = 'assigned_to_id' AND
+                        tc.value IS NOT NULL
+        )")
+    end
+  end
+
+  register_search_advanced_filter(/in:not_assigned/) do |posts|
+    if @guardian.can_assign?
+      posts.where("topics.id NOT IN (
+        SELECT tc.topic_id
+        FROM topic_custom_fields tc
+        WHERE tc.name = 'assigned_to_id' AND
+                        tc.value IS NOT NULL
+        )")
+    end
+  end
+
+  register_search_advanced_filter(/assigned:(.+)$/) do |posts, match|
+    if @guardian.can_assign?
+      user = User.find_by_username(match)
```

Comment:
What happens if `user` can't be found (eg. the username is wrong)?

Reply:
its a user-picker, so if the username is wrong it won't return any user options. I've sent you a screen-recording on meta do have a look😃

Reply:
Sure, but what about if someone manually forges a request with a wrong username?

Reply:
There's an error in the backend but the front end handles it 
![Screenshot 2020-08-14 at 5 01 36 PM](https://user-images.githubusercontent.com/35112518/92230311-8aeed800-eec8-11ea-8eef-5fdb0c81944f.png)

Reply:
Would be even better if there was no error in the backend 😉